### PR TITLE
76: Add job reference and use it to determine output_esdl name.

### PR DESCRIPTION
**Pull request by lfse-slafleur**

### CODE REVIEW

File: src/omotes_sdk/internal/worker/worker.py [hunk 1]
```diff
@@ -203,13 +217,27 @@ def wrapped_worker_task(
...
+    output_esh = pyesdl_from_string(output_esdl)
+    output_energy_system: EnergySystem = output_esh.energy_system
+    input_esh = pyesdl_from_string(input_esdl)
+    input_energy_system: EnergySystem = input_esh.energy_system
+    if job_reference is None:
+        output_energy_system.name = f"{input_energy_system.name}_{WORKER_TASK_TYPE}"
+    elif job_reference == "":
+        output_energy_system.name = f"{input_energy_system.name}"
+    else:
+        output_energy_system.name = f"{input_energy_system.name}_{job_reference}"
```
Comment (MarkTNO) [inline src/omotes_sdk/internal/worker/worker.py:238]: should we replace invalid xml characters with `_` for example?

File: src/omotes_sdk/omotes_interface.py [hunk 2]
```diff
@@ -319,6 +329,8 @@ def submit_job(
         :param auto_disconnect_on_result: Remove/disconnect from all queues pertaining to this job
             once the result is received and handled without exceptions through
             `callback_on_finished`.
+        :param job_reference: An optional reference to the submitted job which is used in the
+            name of the output ESDL as well as in internal logging of OMOTES.
```
Comment (MarkTNO) [inline src/omotes_sdk/omotes_interface.py:333]: Very nice to have it in the internal logging as well. This is to be implemented?

**Review by MarkTNO — APPROVED** [review]

File: src/omotes_sdk/internal/worker/worker.py (hunk 1 above)
Comment (lfse-slafleur) [reply src/omotes_sdk/internal/worker/worker.py:238]: Good catch! Checked what happened when I inserted reserved characters in the job reference. Pyesdl already takes care of the necessary escaping:
```xml
<esdl:EnergySystem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:esdl="http://www.tno.nl/esdl" id="c53b2878-6f0e-49fb-b9a1-9aeb8b808117" description="" name="Untitled EnergySystem with return network_example_sdk_client.simulator %&gt;&gt;&amp;#(@#)(@#$&lt;&gt;)(ASd&lt;&lt;&lt;&gt;&gt;&gt;&gt;&gt;" version="1" esdlVersion="v2207">
```

File: src/omotes_sdk/omotes_interface.py (hunk 2 above)
Comment (lfse-slafleur) [reply src/omotes_sdk/omotes_interface.py:333]: Yes! Originally meant for a more bare implementation to check if the implementation fits everyone. Logging introduced here: https://github.com/Project-OMOTES/omotes-sdk-python/pull/77/commits/e67013198a816de4e2a7c7f06abc21b132b00a0f

**Review by MarkTNO — APPROVED** [review]

**Pull request merged**